hey-discount/src/main/java/Ads.java

```diff
@@ -0,0 +1,121 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class Ads {
+    private List<Ad> ads;
+
+    public Ads(){
+        ads = new ArrayList<>();
+    }
+
+    public Ad getAd(String id){
+        for (Ad ad : ads) {
+            if(ad.getId().equals(id)) {
+                return ad;
+            }
+        }
+        return null;
+    }
+
+    public boolean validate (Ad ad){
+        if (ad.getId() == null || ad.getId().length() < 1) {
```

Comment:
It would be a great deal better to encapsulate this check into `Ad` itself. `if (ad.isValid()) { ... }